src/compiler/checker.ts

```diff
@@ -8409,6 +8449,86 @@ namespace ts {
             return links.resolvedType;
         }
 
+        function getTypeFromImportTypeNode(node: ImportTypeNode): Type {
+            const links = getNodeLinks(node);
+            if (!links.resolvedType) {
+                if (node.isTypeOf && node.typeArguments) { // Only the non-typeof form can make use of type arguments
+                    error(node, Diagnostics.type_arguments_can_only_be_used_in_a_ts_file);
+                    links.resolvedSymbol = unknownSymbol;
+                    return links.resolvedType = anyType;
+                }
+                const argumentType = getTypeFromTypeNode(node.argument);
+                const targetMeaning = node.isTypeOf ? SymbolFlags.Value : SymbolFlags.Type;
+                // TODO: Future work: support unions/generics/whatever via a deferred import-type
+                if (!argumentType || !(argumentType.flags & TypeFlags.StringLiteral)) {
+                    error(node.argument, Diagnostics.Import_specifier_must_be_a_string_literal_type_but_here_is_0, argumentType ? typeToString(argumentType) : "undefined");
+                    links.resolvedSymbol = unknownSymbol;
+                    return links.resolvedType = anyType;
+                }
+                const moduleName = (argumentType as StringLiteralType).value;
+                const innerModuleSymbol = resolveExternalModule(node, moduleName, Diagnostics.Cannot_find_module_0, node, /*isForAugmentation*/ false);
+                if (!innerModuleSymbol) {
+                    error(node, Diagnostics.Cannot_find_module_0, moduleName);
+                    links.resolvedSymbol = unknownSymbol;
+                    return links.resolvedType = anyType;
+                }
+                const moduleSymbol = resolveExternalModuleSymbol(innerModuleSymbol, /*dontResolveAlias*/ false);
+                if (!moduleSymbol) {

```

Comment:
how would this fail? should not always return a symbol?

Reply:
I don't think it can, was probably just being a bit too defensive.